deepchem/models/torch_models/infograph.py

```diff
@@ -0,0 +1,435 @@
+from deepchem.models.torch_models.modular import ModularTorchModel
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn import Sequential, Linear, ReLU, GRU
+from deepchem.feat.graph_data import BatchGraphData
+import math
+
+
+class Encoder(torch.nn.Module):
+    """
+    The encoder for the InfoGraph model. It is a message passing graph convolutional
+    network that produces encoded representations for molecular graph inputs.
+
+    Parameters
+    ----------
+    num_features: int
+        Number of node features for each input
+    edge_features: int
+        Number of edge features for each input
+    dim: int
+        Dimension of the embedding
+    """
+
+    def __init__(self, num_features, edge_features, dim):
+        from torch_geometric.nn import NNConv
+        from torch_geometric.nn.aggr import Set2Set
+
+        super(Encoder, self).__init__()
+        self.lin0 = torch.nn.Linear(num_features, dim)
+
+        nn = Sequential(Linear(edge_features, 128), ReLU(),
+                        Linear(128, dim * dim))
+        self.conv = NNConv(dim, dim, nn, aggr='mean', root_weight=False)
+        self.gru = GRU(dim, dim)
+
+        self.set2set = Set2Set(dim, processing_steps=3)
+
+    def forward(self, data):
+        out = F.relu(self.lin0(data.node_features))
+        h = out.unsqueeze(0)
+
+        for i in range(3):
+            m = F.relu(self.conv(out, data.edge_index, data.edge_features))
+            out, h = self.gru(m.unsqueeze(0), h)
+            out = out.squeeze(0)
+            feat_map = out
+
+        out = self.set2set(out, data.graph_index)
+        return out, feat_map
+
+
+class FF(nn.Module):
```

Comment:
Can we reuse one of the existing feed forward nn.modules? I think we have a few

Reply:
It is a feedforward with a skip connection, which does not match any of the current FF layers